storage/spanstore/utils.go

```diff
@@ -0,0 +1,17 @@
+package spanstore
+
+import (
+	"hash"
+	"math"
+)
+
+// HashBytes returns the uint64 hash value of bytes slice
+func HashBytes(h hash.Hash64, bytes []byte) uint64 {
```

Comment:
no need to make this a public function or a separate file, is there?

Reply:
I notice `composite.go` and `composite_test.go` belong to different packages so I'm copying that logic to downsampling_writer.go and downsampling_writer_test.go. Even if I move the hash function inside DownSamplingWriter, test file won't have access to it if name starts with lowercase? So I guess uppercase is necessary? Please advise.

Reply:
not sure I follow, in golang, private objects are package scoped. Since both `downsampling_writer.go` and `downsampling_writer_test.go` are in the same package, `hashBytes` being private is fine.

Reply:
do you mean package name should not be `spanstore_test` for `composite_test.go` ?

Reply:
AFAIK you use the `_test` suffix in the package name if you ONLY want to test the public interfaces. I don't really see a point for `downsampling_writer.go`

Reply:
Got it.